lib/sequent/core/helpers/matchers/class_matcher.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Sequent
+  module Core
+    module Helpers
+      ClassMatcher = Struct.new(:expected_class, keyword_init: true) do
```

Comment:
Why not use a normal class?

```ruby
class ClassMatcher
  def initialize(expected_class:)
  end
  def matches_message?(message)
     message.is_a?(expected_class)
   end
end
```

Reply:
Structs have the advantage of 'by value' comparison. They are used as keys in a hash, so existing values are overwritten by default. For example:

```
on is_a(MyModule), is_a(MyModule) do |message|
  # ...
end
```

Would result in only a single entry in the hash.